dist/c/include/qiskit/complex.h

```diff
@@ -0,0 +1,46 @@
+#ifndef QISKIT__COMPLEX_H
+#define QISKIT__COMPLEX_H
+/**
+ * A complex double.
+ *
+ * See also ``qk_complex64_to_native`` and ``qk_complex64_from_native`` to convert
+ * this struct to (or from) a compiler-native complex number representation.
+ */
+typedef struct {
+    /* The real part. */
+    double re;
+    /* The imaginary part. */
+    double im;
+} QkComplex64;
```

Comment:
The type name makes sense given that it's going to/from Rust's `Complex64`. Still, C doesn't guarantee double to be a 64bit type (though that's the case in on all common platform anyway). So I wonder - do we want to check this? C11 has `_Static_assert` which would be good for cases like this, e.g 
```c
_Static_assert(sizeof(double) == 8, "Qiskit assumes double to be 64 bit");
```
This might be too much and requires us to consider C11 as the minimum standard, so maybe we can just stick with a statement in the docs...

Reply:
If `double` is not 64 bits, then it is probably no longer FFI-safe and the whole API breaks down anyways, no? If yes we probably should specify this generally.